src/ring_theory/jacobson.lean

```diff
@@ -116,4 +121,113 @@ lemma is_jacobson_iso (e : R ≃+* S) : is_jacobson R ↔ is_jacobson S :=
 
 end is_jacobson
 
+
+section localization
+open localization_map
+variables {y : R} (f : localization_map (submonoid.closure {y} : submonoid R) S)
+
+lemma disjoint_closure_singleton_iff_not_mem {I : ideal R} (hI : I.radical = I) :
+  disjoint ((submonoid.closure {y} : submonoid R) : set R) ↑I ↔ y ∉ I.1 :=
+begin
+  refine ⟨λ h, set.disjoint_left.1 h (submonoid.subset_closure (by simp)), λ h, _⟩,
+  rw [disjoint_iff, eq_bot_iff],
+  rintros x ⟨hx, hx'⟩,
+  cases submonoid.mem_closure_singleton.1 hx with n hn,
```

Comment:
I prefer to write `obtain` instead of `cases` if it doesn't split into multiple goals.
```suggestion
  obtain ⟨n, hn⟩ := submonoid.mem_closure_singleton.1 hx,
```